Modules/SubModule_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -222,18 +223,38 @@ function xASL_wrp_ResampleASL(x)
     elseif round(dim4/2)~=dim4/2
         warning('Odd number of control-label pairs, skipping');
         return;
-    else    
+    else
+        
         % Paired subtraction
         [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
         ASL_im = ControlIm - LabelIm;
-        % Save PWI4D
-        fprintf('%s', [PathPWI4D{iSpace} ', ']);
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, ASL_im, 32, false);
+        % preallocate PWI
+        PWI = zeros(size(ASL_im,1),size(ASL_im,2),size(ASL_im,3));
+        
         % Average PWI
-        PWI = xASL_stat_MeanNan(ASL_im, 4);
-        % Save PWI
-        fprintf('%s\n', PathPWI{iSpace});
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI{iSpace}, PWI, 32, false);
+        if isfield(x.modules.asl,'bMultiPLD') && x.modules.asl.bMultiPLD 
+            % multiPLD PWI
+            RepeatsInPLD = size(ASL_im,4)/length(x.Q.Initial_PLD); % we still need to create a single Repeats in X struct
+            for PLDnmbr = 1:length(x.Q.Initial_PLD)

```

Comment:
What does `nmbr` stand for? `nPLD` would be a more understood abbreviated. Always either write it out fully, or use something that is used a lot. also `i` for Index (not `loc`)

Reply:
Done